### PR TITLE
Build report endpoint

### PR DESCRIPTION
**Pull request by bndnio**

### CODE REVIEW

File: report/models.py [hunk 1]
```diff
@@ -10,22 +14,45 @@ class TrackerStatus(Enum):
...
-    animal_id: models.CharField(max_length=255)
-    status: models.CharField(
-        max_length=3, choices=[(tag, tag.value) for tag in TrackerStatus]
+class Tracker(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    animal_id = models.CharField(max_length=255, unique=True)
+    status = models.CharField(
+        max_length=3, choices=[(tag.name, tag.value) for tag in TrackerStatus]
+    )
+    desired_accuracy = models.DecimalField(max_digits=5, decimal_places=1)
```
Comment (Wyllan) [inline report/models.py]: Accuracy is kind of confusing. Maybe should mention radius instead?

File: report/models.py [hunk 2]
```diff
@@ -10,22 +14,45 @@ class TrackerStatus(Enum):
...
-class Record(models):
+class Record(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     tracker = models.ForeignKey(Tracker, on_delete=models.CASCADE)
     timestamp = models.DateTimeField("date time recorded")
-    geo__lat = models.DecimalField(max_digits=9, decimal_places=6)
-    geo__long = models.DecimalField(max_digits=9, decimal_places=6)
+    geo_lat = models.DecimalField(max_digits=9, decimal_places=6)
+    geo_long = models.DecimalField(max_digits=9, decimal_places=6)
+    geo_accuracy = models.DecimalField(max_digits=5, decimal_places=1)
```
Comment (Wyllan) [inline report/models.py]: Same deal as above?

File: report/views.py [hunk 3]
```diff
@@ -1,6 +1,92 @@
...
+    try:
+        record = Record.objects.get(pk=record_id)
+    except Record.DoesNotExist:
+        return HttpResponse(status=404)
+
+    if request.method == "GET":
+        serializer = RecordSerializer(record)
+        return JsonResponse(serializer.data)
+
+    elif request.method == "PUT":
```
Comment (Wyllan) [inline report/views.py:82]: We don't want records to be change so this can probably be removed.

File: report/views.py [hunk 4]
```diff
@@ -1,6 +1,92 @@
...
+
+    if request.method == "GET":
+        serializer = TrackerSerializer(tracker)
+        return JsonResponse(serializer.data)
+
+    elif request.method == "PUT":
+        data = JSONParser().parse(request)
+        serializer = TrackerSerializer(tracker, data=data)
+        if serializer.is_valid():
+            serializer.save()
```
Comment (Wyllan) [inline report/views.py:46]: Verify that the data is actually overwritten

File: report/models.py (hunk 1 above)
Comment (bndnio) [reply report/models.py]: changing to `max_error_radius`

File: report/models.py (hunk 2 above)
Comment (bndnio) [reply report/models.py]: will be `geo_error_radius`

**Review by Wyllan — APPROVED** [review]
loooooks good.

**Pull request merged**